### PR TITLE
feat: add azure-identity .NET example

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

**Requirements**

- [ ] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

part of #187 

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [ ] no

**Notes for Reviewers**:

### CODE REVIEW

File: examples/azure-identity/dotnet/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM --platform=linux/amd64 mcr.microsoft.com/dotnet/sdk:5.0 AS builder
```
Comment (sozercan) [inline examples/azure-identity/dotnet/Dockerfile:1]: looks like 6.0 is LTS?

File: examples/azure-identity/dotnet/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+WORKDIR /app
+ADD . .
+RUN dotnet build akvdotnet.csproj && dotnet publish -c release
+
+ARG BASEIMAGE
+FROM ${BASEIMAGE:-mcr.microsoft.com/dotnet/runtime:5.0}
```
Comment (sozercan) [inline examples/azure-identity/dotnet/Dockerfile:7]: 6.0?

File: examples/azure-identity/dotnet/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+WORKDIR /app
+COPY --from=builder /app/bin/release/netcoreapp5.0/publish/ .
```
Comment (sozercan) [inline examples/azure-identity/dotnet/Dockerfile:9]:
```suggestion
COPY --from=builder /app/bin/release/netcoreapp6.0/publish/ .
```

File: examples/azure-identity/dotnet/Makefile [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+container-linux:
+	docker buildx build \
+		--output=$(OUTPUT_TYPE) \
+		--platform="linux/$(ARCH)" \
+		--tag=$(DEMO_IMAGE)-linux-$(ARCH) .
+
+.PHONY: container-windows
+container-windows:
+	docker buildx build \
+		--build-arg BASEIMAGE=mcr.microsoft.com/dotnet/runtime:5.0-nanoserver-${OSVERSION} \
```
Comment (sozercan) [inline examples/azure-identity/dotnet/Makefile:37]:
```suggestion
		--build-arg BASEIMAGE=mcr.microsoft.com/dotnet/runtime:6.0-nanoserver-${OSVERSION} \
```

File: examples/azure-identity/dotnet/akvdotnet.csproj [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp5.0</TargetFramework>
```
Comment (sozercan) [inline examples/azure-identity/dotnet/akvdotnet.csproj:5]:
```suggestion
    <TargetFramework>netcoreapp6.0</TargetFramework>
```

File: examples/azure-identity/dotnet/Dockerfile (hunk 1 above)
Comment (aramase) [reply examples/azure-identity/dotnet/Dockerfile]: updated all references to `6.0`

**Review by sozercan — APPROVED** [review]
LGTM

**Review by sozercan — APPROVED** [review]

**Pull request merged**